examples/jsm/controls/OrbitControls.js

```diff
@@ -740,67 +740,91 @@ class OrbitControls extends EventDispatcher {
 
 			let needsUpdate = false;
 
-			switch ( event.code ) {
+			if ( scope.enablePan === true ) {
 
-				case scope.keys.UP:
+				switch ( event.code ) {
 
-					if ( event.ctrlKey || event.metaKey || event.shiftKey ) {
+					case scope.keys.UP:
 
-						rotateUp( 2 * Math.PI * scope.rotateSpeed / scope.domElement.clientHeight );
+						if ( event.ctrlKey || event.metaKey || event.shiftKey ) {
 
-					} else {
+							rotateUp( 2 * Math.PI * scope.rotateSpeed / scope.domElement.clientHeight );
 
-						pan( 0, scope.keyPanSpeed );
+						} else {
 
-					}
+							pan( 0, scope.keyPanSpeed );
 
-					needsUpdate = true;
-					break;
+						}
 
-				case scope.keys.BOTTOM:
+						needsUpdate = true;
+						break;
 
-					if ( event.ctrlKey || event.metaKey || event.shiftKey ) {
+					case scope.keys.BOTTOM:
 
-						rotateUp( - 2 * Math.PI * scope.rotateSpeed / scope.domElement.clientHeight );
+						if ( event.ctrlKey || event.metaKey || event.shiftKey ) {
 
-					} else {
+							rotateUp( - 2 * Math.PI * scope.rotateSpeed / scope.domElement.clientHeight );
 
-						pan( 0, - scope.keyPanSpeed );
+						} else {
 
-					}
+							pan( 0, - scope.keyPanSpeed );
 
-					needsUpdate = true;
-					break;
+						}
 
-				case scope.keys.LEFT:
+						needsUpdate = true;
+						break;
 
-					if ( event.ctrlKey || event.metaKey || event.shiftKey ) {
+					case scope.keys.LEFT:
 
-						rotateLeft( 2 * Math.PI * scope.rotateSpeed / scope.domElement.clientHeight );
+						if ( event.ctrlKey || event.metaKey || event.shiftKey ) {
 
-					} else {
+							rotateLeft( 2 * Math.PI * scope.rotateSpeed / scope.domElement.clientHeight );
 
-						pan( scope.keyPanSpeed, 0 );
+						} else {
 
-					}
+							pan( scope.keyPanSpeed, 0 );
 
-					needsUpdate = true;
-					break;
+						}
 
-				case scope.keys.RIGHT:
+						needsUpdate = true;
+						break;
 
-					if ( event.ctrlKey || event.metaKey || event.shiftKey ) {
+					case scope.keys.RIGHT:
 
-						rotateLeft( - 2 * Math.PI * scope.rotateSpeed / scope.domElement.clientHeight );
+						if ( event.ctrlKey || event.metaKey || event.shiftKey ) {
 
-					} else {
+							rotateLeft( - 2 * Math.PI * scope.rotateSpeed / scope.domElement.clientHeight );
 
-						pan( - scope.keyPanSpeed, 0 );
+						} else {
 
-					}
+							pan( - scope.keyPanSpeed, 0 );
 
-					needsUpdate = true;
-					break;
+						}
+
+						needsUpdate = true;
+						break;
+
+				}
+
+			}
+
+			if ( scope.enableZoom === true ) {
+
+				switch ( event.key ) {
+
+					case scope.keys.PLUS:
+
+						dollyIn( getZoomScale() );
+						needsUpdate = true;
+						break;
+
+					case scope.keys.MINUS:
+
+						dollyOut( getZoomScale() );
+						needsUpdate = true;
+						break;
+
+				}
```

Comment:
I'm seeing in the demo that it only zooms in when I press "shift + =". Would it be best to use `event.code` and check for `"Minus"` and `"Equals"` so it always works when the appropriate keys are pressed?

Reply:
What happens when you press a single key? Do you mind sharing screenshots of https://www.toptal.com/developers/keycode when you press `+` and `-`?

I don't think we can use `event.code` because plus and minus exists in so many variations.

Reply:
When pressing just the `=` key:

```js
{
 "key": "=",
 "keyCode": 187,
 "which": 187,
 "code": "Equal",
 "location": 0,
 "altKey": false,
 "ctrlKey": false,
 "metaKey": false,
 "shiftKey": false,
 "repeat": false
}
```

When pressing `shift + =`:

```js
{
 "key": "+",
 "keyCode": 187,
 "which": 187,
 "code": "Equal",
 "location": 0,
 "altKey": false,
 "ctrlKey": false,
 "metaKey": false,
 "shiftKey": true,
 "repeat": false
}
```

When pressing the `-` key:

```js
{
 "key": "-",
 "keyCode": 189,
 "which": 189,
 "code": "Minus",
 "location": 0,
 "altKey": false,
 "ctrlKey": false,
 "metaKey": false,
 "shiftKey": false,
 "repeat": false
}
```

I'm on a 2021 M1 Macbook Pro keyboard.

Reply:
Thanks for sharing the data! Overall, I think the result on your computer is as expected. When using `event.key`, the controls react on whatever key or key combination produces a specific value. In this case plus and minus. 

Like mentioned in #26826, there are use cases where a relying on physical key positions is better like when developing a game that requires WASD or arrow keys input. Even if WASD keys are in a different location (e.g. when using AZERTY keyboards), gamers usually play with the same physical keys (see #21055). 

I believe plus and minus needs to be handled differently though because even in context of WASD layouts, plus and minus exists in different variations. So checking for the character via `event.key` should deliver more reasonable results.

Reply:
I didn't think the +/= key moved around all the much (same spots in QWERTY, AZERTY, DVORAK, etc) - though it looks like it's not in the common spot with a QWERTZ layout, at least.

I guess what I was getting at is that the demo seemed like it had an issue because the key didn't work until I tried shift. I think if I used an app that told me to zoom with - and + I'd think it was broken.

The only other option I can think of is supporting both + and = characters - but I understand that might be unacceptable. Just a suggestion.

Reply:
I think the user's hands should be able to remain fixed. Consequently, I'd vote for UP/DOWN in combination with one of the option keys -- if possible.

Reply:
>  I'd vote for UP/DOWN in combination with one of the option keys -- if possible.

I'm afraid this will interfere existing code in `handleKeyDown()`. Using _only_ the arrow keys means panning, but with either meta/shift/ctrl rotation is triggered. E.g.:

https://github.com/mrdoob/three.js/blob/8d71879573ba4b00979d00fed36740f2f9bbce48/examples/jsm/controls/OrbitControls.js#L745-L759